qutebrowser/mainwindow/mainwindow.py

```diff
@@ -458,8 +460,23 @@ def resizeEvent(self, e):
         self._downloadview.updateGeometry()
         self.tabbed_browser.tabBar().refresh()
 
+    def showEvent(self, e):
+        """Extend showEvent to register us as the last-visible-main-window.
+
+        Args:
+            e: The QShowEvent
+        """
+        super().showEvent(e)
+        objreg.register('last-visible-main-window', self, update=True)
+
     def _do_close(self):
         """Helper function for closeEvent."""
+        last_visible = objreg.get('last-visible-main-window')
+        if self == last_visible:
```

Comment:
`==` might work here if Qt doesn't override it, but since you actually want to compare object identities here, `is` is the idiomatic thing to use here.